pynamodb_attributes/unicode_delimited_tuple.py

```diff
@@ -44,12 +45,15 @@ def __init__(
         self.delimiter = delimiter
 
     def deserialize(self, value: str) -> T:
-        fields = getattr(self.tuple_type, "_fields", None)
-        field_types = getattr(self.tuple_type, "_field_types", None)
-        if fields and field_types:
-            values = value.split(self.delimiter, maxsplit=len(fields))
+        type_hints = get_type_hints(self.tuple_type)
+
+        if type_hints:
+            values = value.split(self.delimiter, maxsplit=len(type_hints))
             return self.tuple_type(
-                **{f: field_types[f](v) for f, v in zip(fields, values)}
+                **{
+                    f: self._parse_value(v, type_hints[f])
+                    for f, v in zip(type_hints, values)
```

Comment:
```suggestion
                    f: self._parse_value(v, f)
                    for f, v in zip(type_hints.values(), values)
```

Reply:
Could this mess up anything in ordering?

Reply:
Also, we need the key name, which is a string, maybe we could have:
```
f: self._parse_value(v, t)
for ((f, t), v) in zip(type_hints.items(), values)
```

Let me know.

Reply:
This works :)